classes/order/Order.php

```diff
@@ -691,6 +692,21 @@ protected function setProductCustomizedDatas(&$product, $customized_datas)
         }
     }
 
+    /**
+     * If order line targets a combination, replace row stock location data
+     * by combination stock location data.
+     *
+     * @param array &$product
+     */
+    protected function setProductStockLocationForCombination(&$product)
+    {
+        if ($product['product_attribute_id'] === '0') {
+            return;
+        }
+        $combination = new Combination($product['product_attribute_id']);
+        $product['location'] = $combination->location;
```

Comment:
can `$product['location']`not be empty? combination overrides its value then?

Reply:
In my understanding, the data is either carried by the `product` object or the `combination` object, not both. I tried several usecases and it seemed to fit this theory but I cannot assure it to be 100% right.

Reply:
that's what I thought too, just wanted to be sure ;)